pkgs/development/tools/pre-commit/default.nix

```diff
@@ -0,0 +1,28 @@
+{ stdenv, buildPythonApplication, fetchPypi, aspy-yaml, cached-property, cfgv, identify, nodeenv, six, toml, virtualenv}:
+buildPythonApplication rec {
+  pname = "pre_commit";
+  version = "1.10.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1kn8h9k9ca330m5n7r4cvxp679y3sc95m1x23a3qhzgam09n7jwr";
+  };
+
+  propagatedBuildInputs = [
+    aspy-yaml
+    cached-property
+    cfgv
+    identify
+    nodeenv
+    six
+    toml
+    virtualenv
+  ];
+
+  meta = with stdenv.lib; {
+    description = "A framework for managing and maintaining multi-language pre-commit hooks.";
+    homepage = https://pre-commit.com/;
+    license = licenses.mit;
+    maintainers = with maintainers; [ borisbabic ];
```

Comment:
You need to add yourself to https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix. Please do so in an additional commit.